### PR TITLE
Auth functionalities

### PR DESCRIPTION
**Pull request by Mohitk391**
Authentication functionalities added.

Files to check: 
1) Login.js
2) Signup.js
3) NavBar.js
4) UserContext.js
5) UserReducer.js
6) Cart.js
7) CartReducer.js
8) VerticalCard.js

preview:  https://62d526a0c1d65d13f185c1b6--easyui-cl.netlify.app/

### CODE REVIEW

**Review by Ankur9669 — APPROVED** [review]
Code looks fine to me

File: src/components/NavBar/NavBar.js [hunk 1]
```diff
@@ -8,20 +8,30 @@ function NavBar() {
...
                 <Link to="/" ><i className="fa-solid fa-magnifying-glass"></i></Link>
             </div>
+            { userState.isUserLoggedIn ? 
             <div className="nav-bar-buttons">
-                {!userState.isUserLoggedIn ? <Link to="/login"><button className="btn btn-hover sign-in">Login</button></Link> : <button className="btn btn-hover sign-in" onClick={()=>{userDispatch({type: "UNSET_USER"})}}>Logout</button>}
-                <Link to="/wishlist" className="nav-bar-icon badge-container"><i className="fa fa-heart-o" aria-hidden="true"></i><span className="shopping-badge">{wishlistState.itemsInWishlist}</span></Link>
-                <Link to="/cart" className="nav-bar-icon badge-container"><i className="fa fa-cart-plus" aria-hidden="true"></i><span className="shopping-badge">{items}</span></Link>
-                <Link to="/" className="nav-bar-icon"><i className="fa fa-user" aria-hidden="true"></i></Link>
+                <button className="btn btn-hover sign-in" onClick = {()=>{localStorage.removeItem('token');userDispatch({type:"UNSET_USER"}); navigate("/login")}}>Logout</button>
```
Comment (MayankKumar10) [inline src/components/NavBar/NavBar.js:22]: You can write all the logic in a function and call it on the button onClick call. You can Pass localStorage.clear() to clear data in localStorage.

File: src/pages/cart/Cart.js [hunk 2]
```diff
@@ -2,29 +2,116 @@ import { NavBar } from "../../components/NavBar/NavBar";
...
+                        authorization : token
+                    },
+                }
+            ) : await axios.delete(`/api/user/wishlist/${product._id}`,
+                {
+                    headers: {
+                        authorization : token
+                    },
+                }
+            );
```
Comment (MayankKumar10) [inline src/pages/cart/Cart.js:97]: look into indentation here

**Review by MayankKumar10 — APPROVED** [review]
Look into these Suggestions everything else looks good to me you can merge it.

**Pull request merged**